tests/unit/drivers/test_recursive_traversal.py

```diff
@@ -198,3 +198,43 @@ def test_adjacency1_granularity1():
     assert len(docs[0].matches[0].matches) == DOCUMENTS_PER_LEVEL
     assert len(docs[0].matches[0].matches[0].chunks) == DOCUMENTS_PER_LEVEL
 
+
+def test_adjacency_on_chunks():
+    docs = build_docs()
+    doc = docs[0]
+    doc.ClearField('matches')
+    docs = [doc, ]
+    driver = SliceQL(
+        start=0,
+        end=1,
+        adjacency_range=(1, 1),
+        granularity_range=(1, 2),
+        recur_on=["chunks", ]
+    )
+    driver._traverse_apply(docs)
+    assert len(docs) == 1
```

Comment:
Because of the upper `doc = docs[0]`, I wondered how this can be 1 and not 2.

Reply:
`docs=[doc, ]` at line 206